robyn/__init__.py

```diff
@@ -67,7 +67,7 @@ def _add_route(self, route_type, endpoint, handler, is_const=False):
         """
         return self.router.add_route(route_type, endpoint, handler, is_const)
 
-    def before_request(self, endpoint: str) -> Callable[..., None]:
+    def before_request(self, endpoint: str | None = None) -> Callable[..., None]:
```

Comment:
@AntoineRR , we will have to use the `Union` syntax as we need to support python3.8 onwards

Reply:
True, I always forget this, will fix it soon